CONTRIBUTING.md

```diff
@@ -0,0 +1,333 @@
+
+### Codeblocks
+
+Codeblocks are defined using triple backticks followed by the language wrapped in square brackets, for example: ` ```[javascript] `. They are closed using triple backticks. Codeblocks are grouped into tabs when they follow one another with a single empty line between them:
+
+```plaintext
+    ```[javascript]
+    const channel = realtime.channels.get('channelName');
+    ```
+    ```[objc]
+    ARTRealtimeChannel *channel = [realtime.channels get:@"channelName"];
+    ```
+```
+
+To use nested codeblocks when describing features that are available to the realtime and REST interfaces, use the the same syntax as a normal codeblock but prefix the languages with `realtime_` or `rest_`. Spacing should be continuous as with a normal codeblock, however the snippets for each interface should all be written together, rather than interspersed:
+
+```plaintext
+    ```[realtime_javascript]
+    const channel = realtime.channels.get('channelName');
+    ```
+    ```[realtime_objc]
+    ARTRealtimeChannel *channel = [realtime.channels get:@"channelName"];
+    ```
+
+    ```[rest_javascript]
+    const channel = rest.channels.get('channelName');
+    ```
+    ```[rest_objc]
+    ARTRestChannel *channel = [realtime.channels get:@"channelName"];
+    ```
+```
+
+### In-line code
+
+In-line code should be written between `@` symbols. For example, `the @get()@ method`.
+
+### Variables
+
+The following variables can be inserted into codeblocks to generate values at runtime:
+
+| Variable | Description |
+| -------- | ----------- |
+| `{{API_KEY}}` | Injects a demo API key into the code sample if a user isn't logged in. Enables users to select an API key from from their dashboard if they are logged in. |
+| `{{RANDOM_CHANNEL_NAME}}` | Generates a random channel name. |
+
+### Definition lists
+
+Definition lists are used to display a list of terms and their descriptions, such as the available capabilities.
+
+They use the following syntax:
+
+```plaintext
+- publish := can publish messages to channels
+- presence := can register presence on a channel (enter, update and leave)
+```
+
+### Tables
+
+Textile tables are supported which are slightly more flexible than Markdown ones. They don't need to use a row of hyphens to separate the header and content either.
+
+For the header row use the following syntax:
+
+```plaintext
+|_. Flag |_. Description |
+| SUBSCRIBE | Can subscribe to receive messages on the channel. |
+```
+
+To make a cell span two columns:
+
+```plaintext
+|_/2=. Limit |
+|_. Soft |_. Hard |
+```
+
+### Admonitions
+
+There are three types of admonition that can be used; `note`, `important` and `further-reading`. Update the value of `data-type` to switch between them. Admonitions are written using the HTML `<aside>` tag. Content must be constructed in HTML, other than links and in-line code styling which accept textile format.
```

Comment:
When should you use note VS and when should you use important?

For example... Use important for critical details that affect implementation, whereas note is for additional context.??